desk/app/diary.hoon

```diff
@@ -99,11 +99,16 @@
     =.  cor
       (emil (drop load:epos))
     =/  diaries  ~(tap in ~(key by shelf))
+    =.  cor  
+      %+  roll
+        ~(tap in (~(gas in *(set ship)) (turn diaries head)))
+      |=  [=ship cr=_cor]
```

Comment:
No need to `gas:in` if you're gonna `tap:in` right after. Also, you can do away with the extra `turn` iteration for `head` by implicitly discarding the tail in the `roll`:

```suggestion
      %+  roll  diaries
      |=  [[=ship *] cr=_cor]
```

Reply:
I think `tap:in` after `gas:in` is to get rid of duplicate ships if you're subscribed to multiple things from a single ship.

Reply:
Yeah that's correct trying to avoid sending extraneous cards here that will thrash unsub/resub to the same person over and over. If there's a more concise way to do this I would love that lol.

Reply:
Oh duh, that makes sense. Good catch!